src/search/ipfs-search-table/IPFSSearchTable.js

```diff
@@ -0,0 +1,153 @@
+  }
+
+  const typesQueryLookup = {
+    audio: 'audio',
+    image: 'images',
+    video: 'video',
+    text: 'text',
+    html: 'text',
+    pdf: 'text',
+    directory: 'directories'
+  }
+
+  const getQueryFileType = (type, title, mimetype) => {
+    if (type === 'directory') {
+      return 'directories'
+    }
+    if (type === 'unknown') {
+      return 'other'
+    }
+
+    let strippedTitle = new DOMParser().parseFromString(title, 'text/html')
+    strippedTitle = strippedTitle.body.textContent || ''
+
+    const fileTypeFromExt = typeFromExt(strippedTitle)
+    const queryTypeString = typesQueryLookup[mimetype?.split('/')[0]] || typesQueryLookup[fileTypeFromExt] || 'other'
+    return queryTypeString
+  }
+
+  const thumbnailCellRenderer = ({ rowData }) => {
+    const { title, hash, type, mimetype } = rowData
+    const queryFileType = getQueryFileType(type, title, mimetype)
+    const ipfsSearchDetailBaseUrl = 'https://ipfs-search.com/#/search/detail'
+    const detailLink = `${ipfsSearchDetailBaseUrl}/${queryFileType}/${hash}?q=${hash}&page=1`
+    return (
+      <a href={detailLink} target='_blank' rel='noopener noreferrer' style={{ width: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
+        <FileIcon name={title} type={type} />
+      </a>
+    )
+  }
+
+  const detailViewHeader = () => {
+    return (
+        <IPFSSearchLogoHex />
+    )
+  }
+
+  const exploreViewCellRender = ({ rowData }) => {
+    const { hash } = rowData
+    const exploreDetailUrl = `/#/explore/${hash}`
+
+    return (
+      <a href={exploreDetailUrl} rel="noopener noreferrer" style={{ width: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
+        <StrokeIpld />
+      </a>
+    )
+  }
+
+  const dateViewCellRender = ({ rowData }) => {
+    const lastSeen = rowData['last-seen']
+    const lastSeenDate = new Date(lastSeen)
+    const formattedDate = lastSeen ? `${lastSeenDate.toLocaleDateString()} ${lastSeenDate.toLocaleTimeString()} ` : 'N/A'
+    return (
+      <span>{formattedDate}</span>
+    )
+  }
+
+  const titleCellRender = ({ rowData }) => {
+    const title = rowData?.title || rowData?.metadata?.metadata?.title?.[0] || rowData?.metadata?.metadata?.resourceName?.[0] || rowData?.references?.[0]?.name || ''
+    let strippedTitle = new DOMParser().parseFromString(title, 'text/html')
+    strippedTitle = strippedTitle.body.textContent || ''
+
+    return (
+      <span>{strippedTitle}</span>
+    )
+  }
+
+  const authorCellRender = ({ rowData }) => {
+    const author = rowData?.author || rowData?.metadata?.metadata?.Author?.[0] || ''
+
+    return (
+      <span>{author}</span>
+    )
+  }
+
+  const sizeCellRender = ({ rowData }) => {
+    const { size } = rowData
+    const fileSizeKB = size ? Number(size / 1000).toFixed(2) : null
+    const fileSizeString = fileSizeKB ? `${fileSizeKB} kB` : ''
+
+    return (
+      <span>{fileSizeString}</span>
+    )
+  }
+
```

Comment:
I think this code could be cleaned up a lot if the `CellRender` functional components were pulled into separate files